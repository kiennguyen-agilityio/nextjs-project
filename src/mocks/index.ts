export const categoryOptions = [
  { value: 'option1', label: 'Option 1' },
  { value: 'option2', label: 'Option 2' },
  { value: 'option3', label: 'Option 3' },
];

export const listCategory = [
  {
    id: '1',
    label: 'Email',
    options: [
      { value: 'email1', label: 'email1@example.com' },
      { value: 'email2', label: 'email2@example.com' },
    ],
  },
  {
    id: '2',
    label: 'Full Name',
    options: [
      { value: 'john_doe', label: 'John Doe' },
      { value: 'jane_doe', label: 'Jane Doe' },
    ],
  },
  {
    id: '3',
    label: 'Role',
  },
  {
    id: '4',
    label: 'Joined',
    options: [
      { value: '2021', label: '2021' },
      { value: '2022', label: '2022' },
    ],
  },
];
