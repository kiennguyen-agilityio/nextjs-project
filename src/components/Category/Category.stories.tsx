import type { Meta, StoryObj } from '@storybook/react';

// components
import Category from '@/components/Category';

// mocks
import { roles } from '@/mocks';

const meta = {
  title: 'Components/Category',
  component: Category,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Category>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    roles: roles,
  },
};
