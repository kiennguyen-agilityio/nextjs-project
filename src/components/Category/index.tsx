// mocks
import { listCategory } from '@/mocks';

// components
import Dropdown from '@/components/common/Dropdown';

const Category = () => (
  <thead>
    <tr className="flex items-center justify-between pl-0 md:pl-0 pr-8 md:pr-12 border-b border-gray-200">
      {listCategory.map(({ id, label, options = [] }) => (
        <Dropdown key={id} label={label} options={options} />
      ))}
      <th className="text-left flex items-center mb-4 sm:mb-0" />
    </tr>
  </thead>
);

export default Category;
