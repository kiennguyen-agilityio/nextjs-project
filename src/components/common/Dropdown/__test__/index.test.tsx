import { render } from '@testing-library/react';

// components
import Category from '@/components/common/Dropdown';

// mocks
import { categoryOptions } from '@/mocks';

test('renders dropdown button with label and options', () => {
  const { asFragment } = render(
    <Category label="Category" options={categoryOptions} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
