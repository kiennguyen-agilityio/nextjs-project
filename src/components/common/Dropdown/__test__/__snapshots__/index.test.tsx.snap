// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dropdown button with label and options 1`] = `
<DocumentFragment>
  <div
    class="relative text-left flex items-center sm:w-1/5 mb-4 sm:mb-0 flex-1 min-w-[120px]"
  >
    <select
      class="inline-flex justify-center w-full rounded-3xl px-4 py-2 text-sm font-medium text-gray-700 hover:bg-gray-200 focus:outline-none max-w-32"
    >
      <option
        value=""
      >
        Dropdown
      </option>
      <option
        value="option1"
      >
        Option 1
      </option>
      <option
        value="option2"
      >
        Option 2
      </option>
      <option
        value="option3"
      >
        Option 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
